### PR TITLE
feat: Add inline alert component

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**

The goal of this PR is to create a new inline alert component for the new confirmation alerts based on [figma](https://www.figma.com/file/yUczlH2HbHaHsYijiONKx0/Warnings%2C-Alerts-and-Flags?type=design&node-id=1208%3A25406&mode=design&t=rdnfd7U3palOjUVT-1).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23513?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2242

## **Manual testing steps**

Component not used yet.

### **After**
Informative
image

Non-Critical
image

Critical
image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/components/alerts/inline-alert/inline-alert.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+export default function InlineAlert({
+  onClick,
+  severity = Severity.Info,
+}: InlineAlertProps) {
+  const t = useI18nContext();
+
+  const severityBackground = getSeverityBackground(severity);
+
+  return (
+    <div>
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply ui/pages/confirmations/components/alerts/inline-alert/inline-alert.tsx]: Done https://github.com/MetaMask/metamask-extension/pull/23513/commits/2b5818a9bcd1addde89c879467f5cf83d44fee43 .

File: ui/pages/confirmations/components/alerts/inline-alert/inline-alert.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  }
+}
+
+export default function InlineAlert({
+  onClick,
+  severity = Severity.Info,
+}: InlineAlertProps) {
+  const t = useI18nContext();
+
+  const severityBackground = getSeverityBackground(severity);
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply ui/pages/confirmations/components/alerts/inline-alert/inline-alert.tsx]: I removed the specified css to use the severityBackground in `backgroundColor` property.

File: ui/pages/confirmations/components/alerts/inline-alert/inline-alert.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  return (
+    <div>
+      <div
+        data-testid="inlineAlert"
+        className={classnames(severityBackground, {
+          'inline-alert': true,
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply ui/pages/confirmations/components/alerts/inline-alert/inline-alert.tsx:56]: You are right, moved to  `<Box>` to leverage the properties and remove as much as possible of custom scss.

File: ui/pages/confirmations/components/alerts/inline-alert/inline-alert.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+}: InlineAlertProps) {
+  const t = useI18nContext();
+
+  const severityBackground = getSeverityBackground(severity);
+
+  return (
+    <div>
+      <div
+        data-testid="inlineAlert"
+        className={classnames(severityBackground, {
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply ui/pages/confirmations/components/alerts/inline-alert/inline-alert.tsx]: Removed `severityBackground` when changing `div` to `Box` as suggested in the other comment.

File: ui/pages/confirmations/components/alerts/inline-alert/index.scss [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  display: inline-flex;
+  align-items: center;
+  gap: 4px;
+  font-size: 12px;
+  max-width: 100%;
+  font-family: Euclid Circular B;
+  font-weight: 400;
+
+
+  &__critical {
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply ui/pages/confirmations/components/alerts/inline-alert/index.scss]: Done https://github.com/MetaMask/metamask-extension/pull/23513/commits/2b5818a9bcd1addde89c879467f5cf83d44fee43 .

File: ui/pages/confirmations/components/alerts/inline-alert/inline-alert.test.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders alert with danger severity', () => {
+    const { container } = render(
+      <InlineAlert onClick={onClickMock} severity={Severity.Danger} />,
+    );
+
+    expect(
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply ui/pages/confirmations/components/alerts/inline-alert/inline-alert.test.tsx]: Removed assert of class names.

File: ui/pages/confirmations/components/alerts/inline-alert/inline-alert.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  args: {
+    onClick: () => {},
+  },
+} as Meta<typeof InlineAlert>;
+
+const Template: StoryFn<InlineAlertProps> = (args) => <InlineAlert {...args} />;
+
+/**
+ * Inline Informative alert. This is the default alert type.<br/>
+ * The component will still display a modal with more information when clicked.
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply ui/pages/confirmations/components/alerts/inline-alert/inline-alert.stories.tsx]: Done. https://github.com/MetaMask/metamask-extension/pull/23513/commits/2b5818a9bcd1addde89c879467f5cf83d44fee43

**Review by OGPoyraz — APPROVED** [review]
Smooth PR!

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by benjisclowder** [comment]
Hi @vinistevam, do we have any idea of when this component will be visible to users, in which release?

**Comment by vinistevam** [comment]
> Hi @vinistevam, do we have any idea of when this component will be visible to users, in which release?

Hey @benjisclowder, I would say probably mid-quarter, but it will be available behind the `conf-redesign` flag. I already have a good part of the work done but it's pending review from the confirmations team.

**Pull request merged**